src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -171,7 +171,18 @@ void SolverInterfaceImpl::configure(
 #ifndef NDEBUG
     PRECICE_INFO("Configuration: Debug");
 #else
-    PRECICE_INFO("Configuration: Release (Debug and Trace log unavailable)");
+    // Assemble and print release message
+    std::string releaseMessage = "Release";
+#ifndef PRECICE_NO_DEBUG_LOG
+    releaseMessage.append(" + debug log");
+#endif
+#ifndef PRECICE_NO_TRACE_LOG
+    releaseMessage.append(" + trace log");
+#endif
+#ifndef PRECICE_NO_ASSERTIONS
+    releaseMessage.append(" + assertions");
+#endif
+    PRECICE_INFO("Configuration: {}", releaseMessage);
 #endif
     PRECICE_INFO("Configuring preCICE with configuration \"{}\"", configurationFileName);
```

Comment:
You are right, this is quite confusing.
I try to avoid `variant` as this would typically include all options (like Spack does).

We could simply stick to the CMake name:

```diff
 This is preCICE version 2.3.0
 Revision info: v2.3.0-102-g1fe91be0-dirty
+Build type: Release + debug log + assertions
-Build type: Debug
 Configuring preCICE with configuration "../precice-config.xml"
 I am participant "Dirichlet"
 Setting up master communication to coupling partner/s
```

This may also clarify the link between the log message and the `CMAKE_BUILD_TYPE` option.

However, currently this does not show that the build log is not available.
Now that we have this information in the code base, we could print a DEBUG message: `Debug information is not available in this build. Please use a Debug build or recompile with the CMake option PRECICE_RELEASE_WITH_TRACE_LOG enabled.`

Reply:
"Build type" sounds indeed appropriate. I guess with the diff here you are trying to show the difference between a Release and a Debug type (the diff confused me a bit as I took it for a suggestion).

I agree with having a message (don't we already have something like this? I also guess it should not be a... debug message, as this would not be available).

Also, why name it `..._TRACE_LOG`? Enabling trace log to enable debug mode would be confusing.

I am probably forgetting some of the context here since there have been a few days, I hope my main points are still understandable.

Reply:
> don't we already have something like this?

We had a line in the hello message, but this needed to be rewritten.
I added a text now for the simple Release. I'll update the main post.

> I also guess it should not be a... debug message, as this would not be available.

The `PRECICE_DEBUG` macros does nothing, but one can still directly invoke `log.debug()`.
Anyhow, this is not necessary now.

> Also, why name it ..._TRACE_LOG? Enabling trace log to enable debug mode would be confusing.

`_DEBUG_LOG` is to enable debug log (trace is still off).
`_TACE_LOG` is to enable the trace log separately, which is a brutal performance hit. This allows to enable the one or the other. This is superuser territory though. No user will ever have to use this.